praw/reddit.py

```diff
@@ -273,8 +276,11 @@ def _prepare_objector(self):
             self._objector.register(kind, klass)
 
     def _prepare_prawcore(self):
-        requestor = Requestor(USER_AGENT_FORMAT.format(self.config.user_agent),
-                              self.config.oauth_url, self.config.reddit_url)
+        requestor = self.requestor
```

Comment:
I also wonder if there is a way we can pass in these default values so that USER_AGENT need not be explicitly set outside of the Reddit instance.

For example, rather than passing in a Requestor instance, or only a Requestor class, one can pass in a requestor class, in addition to keyword arguments to that class (e.g., to explicitly pass in a session).

That maybe would look like:

    reddit = praw.Reddit('sitename', requestor_class=JSONDebugRequestor, requestor_kwargs={'session': custom_http_session})

In fact, `requestor_class` could not be set (default to None) in the above example yet this provides a way to pass down the session.